integrations/http/pattern.go

```diff
@@ -0,0 +1,64 @@
+package http
+
+import (
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+
+	"go.autokitteh.dev/autokitteh/internal/kittehs"
+	"go.autokitteh.dev/autokitteh/sdk/sdkerrors"
+)
+
+func extractKeys(s string) (keys []string, err error) {
+	var (
+		curr   strings.Builder
+		inside bool
+	)
+
+	for _, r := range s {
+		if inside {
+			if r != '}' {
+				curr.WriteRune(r)
+			} else {
+				keys = append(keys, strings.TrimSuffix(curr.String(), "..."))
+				curr.Reset()
+				inside = false
+			}
+		} else if r == '{' {
+			inside = true
+		}
+	}
+
+	if inside {
+		err = sdkerrors.NewInvalidArgumentError("unmatched '{'")
+	}
+
+	return
+}
+
+func MatchPattern(pattern string, req string) (vs map[string]string, err error) {
```

Comment:
Missing documentation: when/how/why is this function used?

Reply:
documented.